packages/core/elasticsearch/core-elasticsearch-server-internal/src/elasticsearch_service.ts

```diff
@@ -97,7 +97,8 @@ export class ElasticsearchService
     const esNodesCompatibility$ = pollEsNodesVersion({
       internalClient: this.client.asInternalUser,
       log: this.log,
-      ignoreVersionMismatch: config.ignoreVersionMismatch,
+      ignoreVersionMismatch:
+        config.ignoreVersionMismatch || this.coreContext.env.cliArgs.serverless === true,
```

Comment:
Using `env.cliArgs.serverless` for now. In https://github.com/elastic/kibana/issues/155912, we should move that flag somewhere else.